crates/rpc/rpc/src/trace.rs

```diff
@@ -341,6 +341,21 @@ where
             }
         }
 
+        // apply after and count to traces if specified, this allows for a pagination style.
+        // only consider traces after
+        let all_traces: Vec<_> = if let Some(after) = after {
+            all_traces.into_iter().skip(after as usize).collect()
+        } else {
+            all_traces
+        };
+
+        // at most, return count of traces
+        let all_traces: Vec<_> = if let Some(count) = count {
+            all_traces.into_iter().take(count as usize).collect()
+        } else {
+            all_traces
+        };
```

Comment:
we can solve this without allocating by using Vec::split_off for after 
and Vec::truncate for count but checking the len of all_traces

Reply:
Thanks for the tips here @mattsse. Good callouts on avoiding `collect` wrt allocation. Have also included a check to ensure that `split_off` does not panic where `after > len`